apps/passport-server/src/services/devconnectPretixSyncService.ts

```diff
@@ -791,6 +737,177 @@ export class DevconnectPretixSyncService {
+
+  public get hasCompletedSyncSinceStarting(): boolean {
+    return this._hasCompletedSyncSinceStarting;
+  }
+
+  public constructor(
+    context: ApplicationContext,
+    pretixAPI: IDevconnectPretixAPI,
+    rollbarService: RollbarService | null,
+    semaphoreService: SemaphoreService
+  ) {
+    this.db = context.dbPool;
+    this.rollbarService = rollbarService;
+    this.semaphoreService = semaphoreService;
+    this.pretixAPI = pretixAPI;
+    this.organizers = new Map();
+    this._hasCompletedSyncSinceStarting = false;
+  }
+
+  public startSyncLoop(): void {
+    logger("[DEVCONNECT PRETIX] Starting sync loop");
+
+    const trySync = async (): Promise<void> => {
+      await this.trySync();
+      this.timeout = setTimeout(
+        () => trySync(),
+        DevconnectPretixSyncService.SYNC_INTERVAL_MS
+      );
+    };
+
+    trySync();
+  }
+
+  public async trySync(): Promise<void> {
+    try {
+      logger("[DEVCONNECT PRETIX] (Re)loading Pretix Config");
+      await this.loadConfiguration();
+
+      logger("[DEVCONNECT PRETIX] Sync start");
+      await this.sync();
+      await this.semaphoreService.reload();
+      this._hasCompletedSyncSinceStarting = true;
+      logger("[DEVCONNECT PRETIX] Sync successful");
+    } catch (e) {
+      this.rollbarService?.reportError(e);
+      logger("[DEVCONNECT PRETIX] Sync failed", e);
+    }
+  }
+
+  public stop(): void {
+    if (this.timeout) {
+      clearTimeout(this.timeout);
+    }
+  }
+
+  /**
+   * (Re)load Pretix configuration, and set up organizers.
+   */
+  private async loadConfiguration(): Promise<void> {
+    const devconnectPretixConfig = await getDevconnectPretixConfig(this.db);
+
+    if (!devconnectPretixConfig) {
+      throw new Error("Pretix Config could not be loaded");
+    }
+
+    const orgIds = new Set(
+      devconnectPretixConfig.organizers.map((org) => org.id)
+    );
+
+    const previousOrgIds = new Set(this.organizers.keys());
+    const removedOrgIds = [...previousOrgIds].filter((x) => !orgIds.has(x));
+    const newOrgIds = [...orgIds].filter((x) => !previousOrgIds.has(x));
+
+    // Make sure we have an OrganizerSync for any new organizers.
+    // This is also how initial OrganizerSyncs are created.
+    for (const id of newOrgIds) {
+      const config = devconnectPretixConfig.organizers.find(
+        (org) => org.id === id
+      ) as DevconnectPretixOrganizerConfig;
+      const org = new OrganizerSync(
+        config,
+        this.pretixAPI,
+        this.rollbarService,
+        this.db
+      );
+      this.organizers.set(id, org);
+    }
+
+    for (const id of removedOrgIds) {
+      // @todo cleanup any organizers which have pending requests?
+      this.organizers.delete(id);
+    }
+  }
+
+  private async syncSingleOrganizer(
+    id: string,
+    organizer: OrganizerSync
+  ): Promise<void> {
+    return traced(NAME, "syncSingleOrganizer", async (span) => {
+      span?.setAttribute("organizers_count", this.organizers.size);
```

Comment:
you should also `setAttribute` for the `orgId` and `orgUrl` here in a similar way that we do it in other places (I can't remember off the top of my head whether I used the naming convention of `org_url` or `orgUrl`, but we should be consistent with that here)

Reply:
We don't have access to the organizer ID here - we're just iterating over a list of objects that refer to organizers, but we can't see their internals.